### ISSUE TITLE
Feature request for encoder-decoder reranker

### ISSUE THREAD

**Issue by ItsukiFujii** [UKPLab/sentence-transformers#3885]
Could `CrossEncoder` support encoder-decoder rerankers such as [[KaLM-Reranker-V1](https://huggingface.co/KaLM-Embedding/KaLM-Reranker-V1-Nano)](https://huggingface.co/KaLM-Embedding/KaLM-Reranker-V1-Nano)?

Currently, generative reranker support only recognizes `*ForCausalLM` models. KaLM uses `T5Gemma2ForConditionalGeneration` with `is_encoder_decoder=true`, so it is incorrectly routed to the sequence-classification path.

KaLM encodes the document with the encoder, processes the instruction and query with the decoder, and calculates relevance from the final `yes`/`no` logits.

Possible support could include:

- An `AutoModelForSeq2SeqLM`-backed Transformer task.
- Automatic detection of encoder-decoder models.
- Separate encoder and decoder inputs.
- Compatibility with `CrossEncoder.predict()` and `rank()`.

```python
model = CrossEncoder("KaLM-Embedding/KaLM-Reranker-V1-Nano")
scores = model.predict(pairs)
```

**Comment by tomaarsen** [comment]
Hello!

I think it might be possible, but the reranker is meant to be used in 2 phases, i.e. first using the encoder to pre-encoder passages and then using the pre-encoded document + the query (+prompt) in the decoder. 
An integration might be possible if we just ignore that and use the model as a standard CrossEncoder (i.e. just 1 'phase'), perhaps without even requiring custom code. Sentence Transformers supports the prompt template that they use, and we can just fill that with the document, system prompt (i.e. "Given a query, retrieve documents that answer the query."), and the query. 

The cleanest way would be to clone https://huggingface.co/KaLM-Embedding/KaLM-Reranker-V1-Nano, and then try to add the required configuration to the repository, like `modules.json` etc. Overall, it seems pretty similar to the causal rerankers that have been integrated recently, e.g. a `Transformer` plus a `LogitScore` (grabs scores from the `yes` / `no` logits). If you're able to match the behaviour from their current `Using transformers` section, then you can probably extend it to the other checkpoints too:
* https://huggingface.co/KaLM-Embedding/KaLM-Reranker-V1-Small
* https://huggingface.co/KaLM-Embedding/KaLM-Reranker-V1-Large

Seems like an interesting challenge.

- Tom Aarsen

**Comment by tomaarsen** [comment]
Resolved as described in #3889, so I'll close this now!

- Tom Aarsen
